example/spec/parallel/dbRecords/DeadLetterArchiveProcessingSpec.js

```diff
@@ -226,13 +276,18 @@ describe('A dead letter record archive processing operation', () => {
   it('deletes the s3 objects corresponding to successfully processed dead letters', async () => {
     if (beforeAllFailed) fail(beforeAllFailed);
     else {
-      await expectAsync(waitForListObjectsV2ResultCount({
-        bucket: systemBucket,
-        prefix: archivePath,
-        desiredCount: 0,
-        interval: 5 * 1000,
-        timeout: 30 * 1000,
-      })).toBeResolved();
+      expect(await s3ObjectExists({ Bucket: systemBucket, Key: messageKey })).toBeFalse();
+    }
+  });
+
+  it('transfers the s3 objects corresponding to unsuccessfully processed dead letters to a new location in s3', async () => {
```

Comment:
So for the unit tests, you have a test for if the transfer of the failing message does not succeed. Is that not necessary here?

Reply:
I'm not sure writing a test like that is possible for an integration test since I am not sure how to make S3 writes fail in this setting 🤔